### ISSUE TITLE
Wasm crash with invalid page limit

### ISSUE THREAD

**Issue by rossjones** [bacalhau-project/bacalhau#2988]
The [original report](https://bacalhauproject.slack.com/archives/C052BAZJVEU/p1699076868497019) came in mixed with output from another job, so this has been rebuilt.

```
panic: memoryLimitPages invalid: 13107200 > 65536

goroutine 153030 [running]:
tetratelabs/wazero.(*runtimeConfig).WithMemoryLimitPages(0xc000999540, 0xc80000)
        tetratelabs/wazero@v1.5.0/config.go:263 +0x105
pkg/executor/wasm.(*Executor).Start(0xc0008d2c20, {0x34b4768, 0xc0023c5530}, 0xc0018a6aa0)
        pkg/executor/wasm/executor.go:85 +0xce7
pkg/compute.(*BaseExecutor).Start(0xc00221e0a0, {0x34b4768, 0xc0023c5530}, 0xc0025bce00)
        pkg/compute/executor.go:247 +0x61a
pkg/compute.(*BaseExecutor).Run(0xc00221e0a0, {0x34b4730, 0xc0023c54d0}, {0xc0025bce00, {0xc000a5d020, 0x2e}, 0x2, 0x2, {0x1d88e8b5, 0xedcce355b, ...}, ...})
        pkg/compute/executor.go:292 +0x472
pkg/compute.(*ExecutorBuffer).doRun.func1()
        pkg/compute/executor_buffer.go:139 +0xb6
created by pkg/compute.(*ExecutorBuffer).doRun
        pkg/compute/executor_buffer.go:138 +0x274
```

It looks like the issue is the calculation at https://github.com/bacalhau-project/bacalhau/blob/main/pkg/executor/wasm/executor.go#L82-L86 but attempts to trigger the same behaviour without the invocation went nowhere on 1.1.3

Have asked for replication steps.

**Comment by Nasfame** [comment]
@rossjones I think the issue here is that the memory limit for the Wasm module is set to 13107200 pages, which is greater than the maximum allowed memory limit of 65536 pages. This is causing the runtime to panic.


TLDR; came across: https://github.com/tetratelabs/wazero/issues/483

**Comment by rossjones** [comment]
@Nasfame Wasm's 32 bit so it can only address 4Gb, so that makes sense.  I was unable to trigger this by asking for more RAM earlier, so was hoping to get steps to replicate so we can be sure we're fixing it in the correct place (e.g. we probably shouldn't wait until that late to check the memory request for WASM).

**Comment by Nasfame** [comment]
@rossjones Is wazero's default runtime 32bit?

**Comment by rossjones** [comment]
> @rossjones Is wazero's default runtime 32bit?

Well, wasm is 32 bit addressing, rather than it being something that wazero imposed.
